source/proxy-dll/component/hashes.cpp

```diff
@@ -250,6 +286,12 @@ namespace hashes
 				load_file(default_file_name_common, HFF_COMMON);
 			}
 		}
+
+		void post_unpack() override
+		{
+			command::add("register_hash", register_hash_f, "Register hash string, Usage: register_hash <hash>");
+			command::add("register_dvar", register_dvar_f, "Register dvar, Usage: register_dvar <name>");
```

Comment:
If you want a particular name, you can give it.

Also, if you know another method to fetch the command parameters without using all this mess, I take it

Reply:
command params are available through a straight forward class inherited from xlabs' handling component;
Example for use here:
```	void register_hash_f(const command::params& params)
	{
		if (params.size() < 2)
		{
			logger::write(logger::LOG_TYPE_INFO, "%s [string]+", args->argv[args->nesting][0]);
			return;
		}

		for (int i = 1; i < this->size(); i++)
		{
			const char* varname = this->get(i);

			hashes::add_hash(fnv1a::generate_hash(varname), varname);
		}
	}
```

Also as quick I read description/codes I didn't notice you adding this to be used from gsc scripts. isn't there any way doing this through gsc methods instead of intermediation of cbuf? or adding new gsc methods are not possible i assume?

Reply:
We can add as many gsc functions/methods as we want, it is just that in gsc the usual pattern is to use cbuf for this kind of dev ui stuff (usually with `devgui_cmd`). The others dev functions are more for high level controls